### PR TITLE
:memo: Add scene parser

### PR DESCRIPTION
**Pull request by Quentintnrl**
Remove all math lib
Use the GLM math lib
Parse the scene file

### CODE REVIEW

File: include/primitives/Properties/abstractProperties.hpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+** EPITECH PROJECT, 2024
+** RTX
+** File description:
+** No file there , just an epitech header example .
+** You can even have multiple lines if you want !
+*/
+
+#pragma once
+
+class abstractProperties {
```
Comment (RenardFute) [inline include/primitives/Properties/abstractProperties.hpp:11]: Please use PascalCase

File: include/primitives/Properties/cubeProperties.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+** File description:
+** No file there , just an epitech header example .
+** You can even have multiple lines if you want !
+*/
+
+#pragma once
+#include "primitives/Properties/abstractProperties.hpp"
+#include <cstddef>
+
+class cubeProperties : public abstractProperties {
```
Comment (RenardFute) [inline include/primitives/Properties/cubeProperties.hpp:13]: Please use PascalCase

File: include/primitives/objectsFactory.hpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+** You can even have multiple lines if you want !
+*/
+
+#pragma once
+#include "Object.hpp"
+#include "json/Json.hpp"
+
+class ObjectsFactory {
+public:
+    static Object createObject(JsonObject *obj);
```
Comment (RenardFute) [inline include/primitives/objectsFactory.hpp:15]: I think your factory is missing some functions like ``void registerObject(std::string name, std::function<Object(AbstractPropersties)> creator)``

File: src/primitives/objectsFactory.cpp
Comment (RenardFute) [inline src/primitives/objectsFactory.cpp:1]: Please use CamelCase in your file names

File: tests/testsSceneParser.cpp [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+** No file there , just an epitech header example .
+** You can even have multiple lines if you want !
+*/
+
+#include <criterion/criterion.h>
+#include <iostream>
+#include "SceneParser.hpp"
+#include "primitives/Cube.hpp"
+
+#define TEST_JSON "scenes/tests/test.json"
```
Comment (RenardFute) [inline tests/testsSceneParser.cpp:14]: Not a good practices since it's not the only file use in this tests

**Review by RenardFute — CHANGES_REQUESTED** [review]
Please rename your files and classes. Also, I'm not a big fan of your implementation of the factory. The design pattern you tried to implement actually requires instantiating a factory, fill it and then use it to create an object. It is not supposed to be prefilled and only have static access.

**Review by aimericdrk — APPROVED** [review]

File: include/primitives/Properties/CubeProperties.hpp [hunk 5]
```diff
@@ -7,15 +7,16 @@
...
+#include "json/Json.hpp"
 
-class cubeProperties : public abstractProperties {
+class CubeProperties : public AbstractProperties {
     public:
-        cubeProperties() = default;
-        explicit cubeProperties(std::size_t size) : _size(size) {};
-        explicit cubeProperties(JsonObject *obj) { _size = obj->getInt("size");};
-        ~cubeProperties() = default;
+        CubeProperties() { _size = 1; };
```
Comment (RenardFute) [inline include/primitives/Properties/CubeProperties.hpp:16]: You should call the super constructor

**Review by RenardFute — CHANGES_REQUESTED** [review]
Good in the overall, just move the factories' registers outside their constructors.
Also can you try to keep the indices and vertices outside the properties, the properties class will be full enough no need to make it messier with all the vertices data

**Review by RenardFute — APPROVED** [review]
Fine for me 👌

**Review by RenardFute — APPROVED** [review]

**Pull request merged**